### PR TITLE
Initial angular velocities

### PR DESCRIPTION
**Pull request by jamie-54**
## Main Changes

Added `initialAngularVelocity` in `main.cpp` which is then subtract  from the `measuredAngularVelocity` to give the `angularVelocityCorrected` which is used for the `controlAction`

## Associated Issues

- Addresses #43

### CODE REVIEW

**Comment by alphaville** [comment]
@jamie-54 This looks good - have you noticed whether this has an effect on the behaviour of the quadcopter?

**Comment by alphaville** [comment]
@jamie-54 Would the following work?

```c++
void initAngularVelocity()
{
  float angularVelocityTemp[3] = {0};
  int numInitAngularVelocity = 10;

  for (int i = 0; i < numInitAngularVelocity; i++)
  {
    ahrs.update();
    ahrs.angularVelocity(angularVelocityTemp);
    initialAngularVelocity[0] += angularVelocityTemp[0];
    initialAngularVelocity[1] += angularVelocityTemp[1];
    initialAngularVelocity[2] += angularVelocityTemp[2];
  }
  initialAngularVelocity[0] /= (float)numInitAngularVelocity;
  initialAngularVelocity[1] /= (float)numInitAngularVelocity;
  initialAngularVelocity[2] /= (float)numInitAngularVelocity;
}
```

File: src/main.cpp [hunk 1]
```diff
@@ -69,6 +70,35 @@ void discardImuMeasurements(size_t numMeasurements = 5000)
...
 
+/**
+ * Determine the initial angular velocity readings of the IMU. 
+ * This value is subtracted from the angular velocity readings of the IMU
+ * to get the correct angular velocity. 
+ * All angular velocities should be zero (at the start) when quadrotor is still.
+ */
+void initAngularVelocity()
+{
+  float AngularVelocityTemp[3] = {0};
```
Comment (alphaville) [inline src/main.cpp:81]: [C++ variables should be camel-case starting with a lowercase letter](https://github.com/QUB-ASL/bzzz/blob/main/CONTRIBUTING.md#31-cc)

Comment (jamie-54) [reply src/main.cpp:81]: oops my bad I'll sort that out

**Comment by jamie-54** [comment]
> @jamie-54 Would the following work?
> 
> ```c++
> void initAngularVelocity()
> {
>   float angularVelocityTemp[3] = {0};
>   int numInitAngularVelocity = 10;
> 
>   for (int i = 0; i < numInitAngularVelocity; i++)
>   {
>     ahrs.update();
>     ahrs.angularVelocity(angularVelocityTemp);
>     initialAngularVelocity[0] += angularVelocityTemp[0];
>     initialAngularVelocity[1] += angularVelocityTemp[1];
>     initialAngularVelocity[2] += angularVelocityTemp[2];
>   }
>   initialAngularVelocity[0] /= (float)numInitAngularVelocity;
>   initialAngularVelocity[1] /= (float)numInitAngularVelocity;
>   initialAngularVelocity[2] /= (float)numInitAngularVelocity;
> }
> ```

I don't see any reason for this not to work, I'll test it on hardware tomorrow and then update.
Then I'll check PR #45

**Review by alphaville — APPROVED** [review]
Looks good

**Pull request merged**